dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/message/MessageImpl.java

```diff
@@ -844,1507 +561,649 @@ public Object get(Object key) {
-        this.securityContext = securityContext;
+    public void setSecurityContext(Object s) {
+        propertyValues[securityContext] = s;
     }
 
     @SuppressWarnings("rawtypes")
     public Collection getInterceptorProviders() {
-        return interceptorProviders == NOT_FOUND ? null : (Collection) interceptorProviders;
+        return (Collection) getFromPropertyArray(interceptorProviders);
     }
     
     @SuppressWarnings("rawtypes")
-    public void setInterceptorProviders(Collection interceptorProviders) {
-        this.interceptorProviders = interceptorProviders;
+    public void setInterceptorProviders(Collection i) {
+        propertyValues[interceptorProviders] = i;
     }
 
     public Object getTemplateParameters() {
-        return templateParameters == NOT_FOUND ? null :  templateParameters;
+        return getFromPropertyArray(templateParameters);
     }
     
-    public void setTemplateParameters(Object templateParameters) {
-        this.templateParameters = templateParameters;
+    public void setTemplateParameters(Object t) {
+        propertyValues[templateParameters] = t;
     }
 
     public void removeContentType() {
-        contentType = NOT_FOUND;
+        propertyValues[contentType] = NOT_FOUND;
     }
     public void removeHttpResponse() {
-        httpResponse = NOT_FOUND;
+        propertyValues[httpResponse] = NOT_FOUND;
     }
     public void removeHttpRequest() {
-        httpRequest = NOT_FOUND;
+        propertyValues[httpRequest] = NOT_FOUND;
+    }
+    
+    private Object getFromPropertyArray(int index) {
+        Object value = propertyValues[index];
+        return value == NOT_FOUND ? null : value;
+    }
+    
+    @Override
+    public int size() {
+        int size = super.size();
+        for (Object o : propertyValues) {
+            if (o != NOT_FOUND) {
+                size++;
+            }
+        }
+        return size;
+    }
+    
+    @Override
+    public void clear() {
+        super.clear();
+        for (int i = 0; i < TOTAL; i++) {
+            propertyValues[i] = NOT_FOUND;
+        }
+    }
+    
+    @Override
+    public boolean isEmpty() {
+        if (!super.isEmpty()) {
+            return false;
+        }
+        
+        for (Object o : propertyValues) {
+            if (o != NOT_FOUND) {
+                return false;
+            }
+        }
+        return true;
+    }
+    
+    @Override
+    public boolean containsValue(Object value) {
+        if (super.containsValue(value)) {
+            return true;
+        }
+        
+        for (Object o : propertyValues) {
+            if (o.equals(value)) {
+                return true;
+            }
+        }
+        
+        return false;
+    }
+    
+    @Override
+    public Object getOrDefault(Object key, Object d) {
+        return getOrDefault((String) key, d);
+    }
+    
+    public Object getOrDefault(String key, Object d) {
+        Object v = super.getOrDefault(key, NOT_FOUND);
```

Comment:
This logic should be like the normal get method in this class.  Not sure why you aren't checking the propertyNames first.  You should use the keymap as well here like in get.   The method should be flipped and the last line be super.getOrDefault(key, d);